pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -45,6 +48,46 @@ func ValidateManagedSeed(managedSeed *seedmanagement.ManagedSeed) field.ErrorLis
 
 	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&managedSeed.ObjectMeta, true, apivalidation.NameIsDNSLabel, field.NewPath("metadata"))...)
 	allErrs = append(allErrs, ValidateManagedSeedSpec(&managedSeed.Spec, field.NewPath("spec"), false)...)
+	allErrs = append(allErrs, ValidateHAConfig(managedSeed)...)
+
+	return allErrs
+}
+
+// ValidateHAConfig validates the HA configuration for the ManagedSeed object.
+func ValidateHAConfig(managedSeed *seedmanagement.ManagedSeed) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	seedSpecFldPath := field.NewPath("spec", "seedTemplate", "spec")
+	// extract the seed seedSpec
+	seedSpec, err := helper.ExtractSeedSpec(managedSeed)
+	if err != nil {
+		return append(allErrs, field.Invalid(seedSpecFldPath, managedSeed.Spec, fmt.Sprintf("failed to extract seed seedSpec from managedSeed.seedSpec %v", err)))
+	}
+
+	multiZonalSeedLabelPath := field.NewPath("metadata", "labels").Key(v1beta1constants.LabelSeedMultiZonal)
+	allErrs = append(allErrs, validateManagedSeedShouldNotHaveBothMultiZoneLabelAndHASpec(managedSeed.ObjectMeta, seedSpec, multiZonalSeedLabelPath)...)
```

Comment:
A cross validation with `managedSeed.ObjectMeta` does not help here because the labels are not merged with the seed that's being created/updated.

You rather need to pass `managedSeed.Spec.SeedTemplate.ObjectMeta` or `managedSeed.Spec.Gardenlet.Config.SeedConfig.ObjectMeta`.

Reply:
changed. PTAL